File: Data Structures & Algorithms/n-th-tribonacci-number/submission-1.cs
public class Solution {
    public int Tribonacci(int n) {
        if(n==0 || n==1)
        return n;
        if(n==2)
        return 1;
        int a=0,b=1;
        int c=1,d=0;
        for(int i=3;i<=n;i++)
        {
            d=a+b+c;
            a=b;
            b=c;
            c=d;

        }
        return d;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
     int n=nums.size();
     vector<int> res(n,1);
     int leftProd=1;
     int rightProd=1;
     for(int i=0;i<n;i++)
     {
        res[i]=leftProd;
        leftProd=leftProd*nums[i];
     }
     for(int i=n-1;i>=0;i--)
     {
        
        res[i]=res[i]*rightProd;
        rightProd=rightProd*nums[i];
     }
     return res;
    }
};
